cmdutil/debug/debug.go

```diff
@@ -71,3 +75,88 @@ func (s *Server) Stop(_ error) {
 
 	close(s.done)
 }
+
+// PProfServer wraps a pprof server.
+type PProfServer struct {
+	logger        logrus.FieldLogger
+	addr          string
+	done          chan struct{}
+	pprofServer   *http.Server
+	profileConfig ProfileConfig
+}
+
+// ProfileConfig holds the configuration for the pprof server.
+type ProfileConfig struct {
+	Addr            string
+	ProfileHandlers map[string]http.HandlerFunc
+}
+
+// NewPProfServer sets up a pprof server with configurable profiling types and returns a PProfServer instance.
+func NewPProfServer(config ProfileConfig, l logrus.FieldLogger) *PProfServer {
+	if config.Addr == "" {
+		config.Addr = "127.0.0.1:9998" // Default port
+	}
+
+	// Create a new HTTP mux for handling pprof routes.
+	mux := http.NewServeMux()
+
+	// Iterate over the handlers and add them to the mux.
+	for profile, handler := range config.ProfileHandlers {
+		if handler != nil {
+			if profile == "mutex" {
+				runtime.SetMutexProfileFraction(2)
+			}
+			mux.HandleFunc("/debug/pprof/"+profile, handler)
```

Comment:
We don't need to raw construction of the handlers like this. The `pprof.Handler` function can construct the handlers for us: https://pkg.go.dev/net/http/pprof#Handler